### PR TITLE
[Consensus] Add BlockValidationState to ChainedHeader ToString()

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

File: src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs [hunk 1]
```diff
@@ -338,7 +342,7 @@ public List<ChainedHeaderBlock> PartialValidationSucceeded(ChainedHeader chained
             if ((chainedHeader.Previous.BlockValidationState != ValidationState.PartiallyValidated) &&
                 (chainedHeader.Previous.BlockValidationState != ValidationState.FullyValidated))
             {
-                this.logger.LogTrace("(-)[PREV_BLOCK_NOT_VALIDATED]:null");
+                this.logger.LogTrace("(-)[PREV_BLOCK_NOT_VALIDATED]:{0}", chainedHeader.Previous);
```
Comment (zeptin) [inline src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs]: I thought our convention was to log the return value of the function in the exit log. Not a big deal, but still.

File: src/Stratis.Bitcoin/Consensus/ConsensusManager.cs [hunk 2]
```diff
@@ -550,6 +550,9 @@ private async Task<ConnectBlocksResult> FullyValidateLockedAsync(ChainedHeader n
                     // This might cause uncontrolled memory changes but big reorgs are not common and a chain will anyway get disconnected when the fork is more then 500 blocks.
                     foreach (ChainedHeaderBlock disconnectedBlock in disconnectedBlocks)
                     {
+                        this.logger.LogTrace("[DISCONNECTED_BLOCK_STATE]{0}", disconnectedBlock.ChainedHeader);
+                        this.logger.LogTrace("[DISCONNECTED_BLOCK_STATE]{0}:{1}:{2}", disconnectedBlock.ChainedHeader.Previous);
```
Comment (zeptin) [inline src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: I think you copy pasted this and didn't update the interpolation :)

Comment (fassadlr) [reply src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: Ah yes, will fix.

File: src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs (hunk 1 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs]: Ok yes I will change :)

File: src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs [hunk 3]
```diff
@@ -591,6 +599,8 @@ public bool BlockDataDownloaded(ChainedHeader chainedHeader, Block block)
             bool partialValidationRequired = chainedHeader.Previous.BlockValidationState == ValidationState.PartiallyValidated
                                           || chainedHeader.Previous.BlockValidationState == ValidationState.FullyValidated;
 
+            this.logger.LogTrace("[BLOCK_DOWNLOAD_PREVIOUS_STATE]{0}:{1}:{2}", nameof(chainedHeader.Previous), chainedHeader.Previous, chainedHeader.Previous.BlockValidationState);
```
Comment (dangershony) [inline src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs]: no need to trace this `chainedHeader.Previous.BlockValidationState` its traced as part of the `Previous` ChainedHeader

**Pull request merged**